scripts/protocolFilters/aerodrome.ts

```diff
@@ -1,7 +1,48 @@
+import { BlockField, QueryResponse } from '@envio-dev/hypersync-client'
 import { ReferralEvent } from '../types'
+import { getBlock, getHyperSyncClient } from '../utils'
+import {
+  AERODROME_NETWORK_ID,
+  AERODROME_UNIVERSAL_ROUTER_ADDRESS,
+} from '../calculateRevenue/protocols/aerodrome/constants'
 
-// TODO(ENG-194): Check that the user has made at least one transaction on Aerodrome,
-// and all transactions were made after the referral event.
 export async function filter(event: ReferralEvent): Promise<boolean> {
-  return !!event
+  const client = getHyperSyncClient(AERODROME_NETWORK_ID)
+  const query = {
+    transactions: [
+      { to: [AERODROME_UNIVERSAL_ROUTER_ADDRESS], from: [event.userAddress] },
+    ],
+    fieldSelection: {
+      block: [BlockField.Number],
+    },
+    fromBlock: 0,
+  }
+
+  let hasMoreBlocks = true
+  let foundValidTransaction = false
+
+  while (hasMoreBlocks) {
```

Comment:
This is the second place we're implementing hypersync pagination, along with in [networks.ts](https://github.com/divvi-xyz/divvi-protocol-v0/blob/main/scripts/calculateRevenue/protocols/utils/networks.ts#L5). I wonder if it's worth it to unify the hypersync requests/pagination into a single helper. If it were something simpler than pagination, I'd say it's fine to keep them duplicated, but since the pagination semantics are potentially a little subtle I wonder if it's worth combining these.

Reply:
Adding in a separate PR: https://github.com/divvi-xyz/divvi-protocol-v0/pull/77.